Sources/MapboxNavigation/NavigationMapView+BuildingHighlighting.swift

```diff
@@ -11,17 +11,96 @@ extension NavigationMapView {
      
      - parameter coordinates: Coordinates which represent building locations.
      - parameter extrudesBuildings: Switch which allows to highlight buildings in either 2D or 3D. Defaults to true.
-     
-     - returns: Bool indicating if number of buildings found equals number of coordinates supplied.
+     - parameter completion: An escaping closure that operates after the buildings are highted.
+
      */
-    @discardableResult public func highlightBuildings(at coordinates: [CLLocationCoordinate2D], in3D extrudesBuildings: Bool = true) -> Bool {
-        return false
+    public func highlightBuildings(at coordinates: [CLLocationCoordinate2D],
+                                   in3D extrudesBuildings: Bool = true,
+                                   completion: @escaping (_ result: Bool) -> Void) {
+        var foundBuildingIds = Set<Double>()
+        let group = DispatchGroup()
+        if let identifiers = try? mapView.__map.getStyleLayers().compactMap({$0.id}).filter({ $0.contains("building") }) {
+            let _ = coordinates.map({
+                group.enter()
+                let screenCoordinate = mapView.point(for: $0, in: self)
+                mapView.visibleFeatures(at: screenCoordinate,
+                                        styleLayers: Set(identifiers),
+                                        completion: { [weak self] result in
+                                            guard let validSelf = self else { return }
```

Comment:
`validSelf` usage produces warning here, because it's not used.

Reply:
Thanks for the suggestion, I'll remove this part.